pkg/storage/cacher.go

```diff
@@ -286,11 +306,26 @@ func (c *Cacher) Watch(ctx context.Context, key string, resourceVersion string,
 }
 
 // Implements storage.Interface.
+func (c *Cacher) WatchList(ctx context.Context, key string, resourceVersion string, filter Filter) (watch.Interface, error) {
+	return c.Watch(ctx, key, resourceVersion, filter)
```

Comment:
@xiang90 - I think that removing this one was actually a problem. But since we were blocked from a lot of time so far, and I didn't have time to really confirm it, I reverted this PR.